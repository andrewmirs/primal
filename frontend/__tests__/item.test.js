import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'ASDF123',
    title: 'A Cool Item',
    price: 4098,
    description: 'This is a realllly cool item',
    image: 'dog.jpg',
    largeImage: 'largedog.jpg',
};

describe('<Item/>', () => {
    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
    
    // it('renders the image properly', () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    //     const img = wrapper.find('img');
    //     expect(img.props().src).toBe(fakeItem.image);
    //     expect(img.props().alt).toBe(fakeItem.title);
    // });

    // it('renders the pricetag and title properly', () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    //     const priceTag = wrapper.find('PriceTag');
    //     // console.log(priceTag.children().text());
    //     expect(priceTag.children().text()).toBe('$50');
    //     // console.log(wrapper.debug());

    //     expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    // });

    // it('renders out the buttons properly', () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    //     const buttonList = wrapper.find('.buttonList');
    //     expect(buttonList.find('Link').exists()).toBe(true);
    //     expect(buttonList.find('AddToCart').exists()).toBe(true);
    //     expect(buttonList.find('DeleteItem').exists()).toBe(true);
    //     console.log(buttonList.debug());
    // });
});