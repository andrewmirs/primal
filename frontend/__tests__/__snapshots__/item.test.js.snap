// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ASDF123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40.98
  </PriceTag>
  <p>
    This is a realllly cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ASDF123",
          },
        }
      }
    >
      <a>
        Edit âœŽ
      </a>
    </Link>
    <AddToCart
      id="ASDF123"
    />
    <DeleteItem
      id="ASDF123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
